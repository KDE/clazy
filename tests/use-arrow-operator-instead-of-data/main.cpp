#include <QtCore/QObject>
#include <QtCore/QScopedPointer>
#include <QtCore/QPointer>
#include <QtCore/QSharedPointer>
class MyObj : public QObject {public: void doStuff();};
void test()
{
    QScopedPointer<QObject> ptr(new QObject);
    ptr.data()->deleteLater(); // Warn

    QPointer<QObject> qptr(new QObject);
    qptr.data()->deleteLater();   // Warn

    QSharedPointer<QObject> qsptr(new QObject);
    qsptr.  data()->deleteLater(); // Warn

    QScopedPointer<QObject> okPtr(new QObject);
    okPtr->deleteLater();

    static_cast<MyObj*>(qsptr.data())->doStuff();
}

