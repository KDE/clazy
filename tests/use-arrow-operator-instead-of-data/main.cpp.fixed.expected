#include <QtCore/QObject>
#include <QtCore/QScopedPointer>
#include <QtCore/QPointer>
#include <QtCore/QSharedPointer>
class MyObj : public QObject {public: void doStuff();};
void test()
{
    QScopedPointer<QObject> ptr(new QObject);
    ptr->deleteLater(); // Warn

    QPointer<QObject> qptr(new QObject);
    qptr->deleteLater();   // Warn

    QSharedPointer<QObject> qsptr(new QObject);
    qsptr->deleteLater(); // Warn

    QScopedPointer<QObject> okPtr(new QObject);
    okPtr->deleteLater();

    static_cast<MyObj*>(qsptr.data())->doStuff();
}

